src/DataFilterer.ts

```diff
@@ -43,32 +35,39 @@ export class DataFilterer {
         const aggVars: any[] = [...this.getUniqueVariables(-1, filterOptions.disagg, filterOptions.metric, filteredData)];
         const dataByAggregate = this.groupDataByDisaggAndThenCompare(filterOptions.compare, filterOptions.disagg, aggVars, filteredData);
 
-
         let datasets: FilteredRow[] = [];
         for (let aggVar of aggVars) {
-            const dataByCompare = dataByAggregate[aggVar];
-            // this is not const in case we need to convert it to a cumulative plot later
-            let summedMetricForDisagg: number[] = compVars.map(function (country: string) {
-                const data: ImpactDataRow[] = dataByCompare[country];
-                if (typeof data !== 'undefined') { // this is necessary to prevent errors when this compare / aggregate combo is empty
-                    return data.map(x => x[filterOptions.metric])
-                        .filter(x => !isNaN(x))
-                        .reduce((acc, x) => acc + x, 0)
-                        .toPrecision(3); // this should possibly be an argument or calculated at run time
-                } else {
-                    return 0;
-                }
-            });
-
+            let summedMetricForDisagg: number[] = this.reduceSummary(dataByAggregate,
```

Comment:
Much easier to read now this has been factored out into a separate function! :+1: